pkg/querier/querier.go

```diff
@@ -88,25 +90,32 @@ func New(cfg Config, distributor Distributor, chunkStore ChunkStore) (storage.Qu
 }
 
 // NewQueryable creates a new Queryable for cortex.
-func NewQueryable(dq, cq storage.Queryable, distributor Distributor) storage.Queryable {
+func NewQueryable(dq, cq storage.Queryable, distributor Distributor, ingesterMaxChunkAge time.Duration) storage.Queryable {
 	return storage.QueryableFunc(func(ctx context.Context, mint, maxt int64) (storage.Querier, error) {
-		dqr, err := dq.Querier(ctx, mint, maxt)
-		if err != nil {
-			return nil, err
+		q := querier{
+			distributor: distributor,
+			ctx:         ctx,
+			mint:        mint,
+			maxt:        maxt,
 		}
 
 		cqr, err := cq.Querier(ctx, mint, maxt)
 		if err != nil {
 			return nil, err
 		}
 
-		return querier{
-			queriers:    []storage.Querier{dqr, cqr},
-			distributor: distributor,
-			ctx:         ctx,
-			mint:        mint,
-			maxt:        maxt,
-		}, nil
+		// Include ingester only if maxt is within 2 times ingester chunk age w.r.t. current time.
+		if maxt >= time.Now().Add(-2*ingesterMaxChunkAge).UnixNano()/1e6 {
```

Comment:
Instead of asking the chunk age and the multiplying by two, why not just ask for a "high watermark" and let the user do the math?